### ISSUE TITLE
Understanding of Unlabeled Losses

### ISSUE THREAD

**Issue by AKASH2907** [soo89/CSD-SSD#15]
Hi,

Thanks for sharing your code. I would like to inquire more about the JSD loss for classification consistency. specifically this line [Link](https://github.com/soo89/CSD-SSD/blob/0466f9670e781e46cac8d96f880bf5645f49d499/train_csd.py#L292)

Why is .detach() specifically used here for computing the losses? 

And, I didn't understand how it impacts when you are detaching either output in each line? If I print requires_grad on those tensors it shows True.
